src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -92,42 +92,59 @@ public void ExecuteCommand(TypeCharCommandArgs args, Action nextCommandHandler,
         transaction?.Complete();
     }
 
-    private SemicolonBehavior BeforeExecuteCommand(bool speculative, TypeCharCommandArgs args, CommandExecutionContext executionContext)
+    private SemicolonBehavior BeforeExecuteCommand(
+        bool speculative,
+        TypeCharCommandArgs args,
+        CommandExecutionContext executionContext)
     {
         if (args.TypedChar != ';' || !args.TextView.Selection.IsEmpty)
-        {
             return SemicolonBehavior.None;
-        }
 
-        var caretOpt = args.TextView.GetCaretPoint(args.SubjectBuffer);
-        if (!caretOpt.HasValue)
-        {
+        if (args.TextView.GetCaretPoint(args.SubjectBuffer) is not { } caret)
             return SemicolonBehavior.None;
-        }
 
         if (!_globalOptions.GetOption(CompleteStatementOptionsStorage.AutomaticallyCompleteStatementOnSemicolon))
-        {
             return SemicolonBehavior.None;
-        }
 
-        var caret = caretOpt.Value;
         var document = caret.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
         if (document == null)
-        {
             return SemicolonBehavior.None;
-        }
 
         var cancellationToken = executionContext.OperationContext.UserCancellationToken;
         var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
         var root = document.GetRequiredSyntaxRootSynchronously(cancellationToken);
 
-        if (!TryGetStartingNode(root, caret, out var currentNode, cancellationToken))
-        {
+        if (!TryGetStartingNode(root, caret, out var tokenOnLeft, out var currentNode, cancellationToken))
             return SemicolonBehavior.None;
+
+        // If the user types `= new;` complete it out to `= new();`
+        if (tokenOnLeft.Kind() is SyntaxKind.NewKeyword &&
+            currentNode is BaseObjectCreationExpressionSyntax
+            {
+                Parent: EqualsValueClauseSyntax
+                {
+                    Parent: VariableDeclaratorSyntax
+                    {
+                        Parent: VariableDeclarationSyntax
+                        {
+                            Parent: FieldDeclarationSyntax or LocalDeclarationStatementSyntax
+                        }
+                    }
+                }
+            })
+        {
+            if (!speculative)
+            {
+                var edit = args.SubjectBuffer.CreateEdit();
+                edit.Insert(caret, "();");
```

Comment:
>edit.Insert(caret, "();"); [](http://example.com/codeflow?start=16&length=26)

Is adding the class name for the new operator an option that can be specified?

Reply:
Chatted offline, didn't consider that scenario important enough unless there is customer feedgack.